docker.go

```diff
@@ -628,26 +628,30 @@ func (c *DockerContainer) StartLogProducer(ctx context.Context) error {
 			Since:      since,
 		}
 
-		ctx, cancel := context.WithTimeout(ctx, time.Second*5)
-		defer cancel()
-
 		r, err := c.provider.client.ContainerLogs(ctx, c.GetContainerID(), options)
 		if err != nil {
-			// if we can't get the logs, panic, we can't return an error to anything
-			// from within this goroutine
-			panic(err)
+			// if we can't get the logs, retry in one second.
+			if ctx.Err() != nil {
+				// context done.
+				return
+			}
+			c.logger.Printf("cannot get logs for container %q: %v", c.ID, err)
+			time.Sleep(1 * time.Second)
```

Comment:
Something like this would work

```suggestion
			select {
			case <-ctx.Done():
				return	
			default:
				c.logger.Printf("cannot get logs for container %q: %v", c.ID, err)
			}
			select {
			case <-ctx.Done():
				return
			case <-time.After(time.Second):
			}
```

Reply:
Done in [9e4b5a1](https://github.com/testcontainers/testcontainers-go/pull/1777/commits/9e4b5a1ca76c94f4918755abcea36197bc8b3e89), thanks!